api/src/wfl/server.clj

```diff
@@ -76,9 +77,22 @@
       wrap-internal-error
       (wrap-json-response {:pretty true})))
 
-(defn notify-watchers [watchers _uuid _exception]
+(defn notify-watchers [watchers uuid exception]
+  "Notify `watchers` with available approaches."
+  ;; FIXME: add permission checks for slack-channel-watchers
   {:pre [(some? watchers)]}
-  (log/info "notifying: " watchers))
+  (let [slack-msg              (format
+                                 (str/join " "
+                                   ["NOTE: WorkFlow Launcher failed to update"
+                                    "a workload %s you are watching! Details"
+                                    "shown below: \n %s"]) uuid exception)
+        notify-with-slack      (fn [slack-channel-watcher]
+                                  (slack/post-message slack-channel-watcher slack-msg))
+        slack-channel-watcher? (fn [watcher]
+                                 (not (util/email-address? watcher)))
+        slack-channel-watchers (filter slack-channel-watcher? watchers)]
+    (log/info "notifying: " slack-channel-watchers)
```

Comment:
I'd use `split-with` to make the separation clearer: 
```clojure
(let [[channels _] (split-with slack-channel? watchers)]
  (run! notify-with-slack channels))
```

https://clojuredocs.org/clojure.core/split-with

Reply:
I also don't this we should be doing this in the "update loop" (or in an exception handler therein) - we should probably defer to another queue/execution service.

Reply:
Does WFL support mixing Slack channels and Emails in the watchers list?

Reply:
The list currently supports emails - my fault for adding the potential for functionality without actual support for it. We'll need to keep that to avoid reitit errors when returning all workloads in the db. Suggest we extend watchers to contain both emails and extra stuff. See my comment below for one way of doing this.